src/main/kotlin/no/skatteetaten/aurora/fergus/service/StorageGridService.kt

```diff
@@ -1,17 +1,50 @@
 package no.skatteetaten.aurora.fergus.service
 
+import kotlinx.coroutines.reactive.awaitFirst
+import kotlinx.coroutines.reactive.awaitFirstOrNull
 import kotlinx.coroutines.reactive.awaitSingle
 import no.skatteetaten.aurora.fergus.FergusException
+import no.skatteetaten.aurora.fergus.controllers.Access
 import no.skatteetaten.aurora.fergus.controllers.AuthorizationPayload
 import org.openapitools.client.api.AuthApi
+import org.openapitools.client.api.ContainersApi
+import org.openapitools.client.api.GroupsApi
+import org.openapitools.client.api.S3Api
+import org.openapitools.client.api.UsersApi
 import org.openapitools.client.model.AuthorizeResponse
+import org.openapitools.client.model.ContainerCreate
+import org.openapitools.client.model.ContainerCreateResponse
+import org.openapitools.client.model.ContainerListResponse
 import org.openapitools.client.model.Credentials
+import org.openapitools.client.model.GetPatchPostPutGroupResponse
+import org.openapitools.client.model.GetPatchPostPutUserResponse
+import org.openapitools.client.model.InlineObject1
+import org.openapitools.client.model.ListGroupsResponse
+import org.openapitools.client.model.ListUsersResponse
+import org.openapitools.client.model.PasswordChangeRequest
+import org.openapitools.client.model.PatchUserRequest
+import org.openapitools.client.model.Policies
+import org.openapitools.client.model.PolicyS3
+import org.openapitools.client.model.PolicyS3Statement
+import org.openapitools.client.model.PostAccessKeyResponse
+import org.openapitools.client.model.PostGroupRequest
+import org.openapitools.client.model.PostUserRequest
+import org.springframework.beans.factory.annotation.Value
 import org.springframework.http.HttpStatus
 import org.springframework.stereotype.Service
 import org.springframework.web.server.ResponseStatusException
+import java.util.UUID
 
 @Service
-class StorageGridServiceReactive(private val storageGridAuthApi: AuthApi) : StorageGridService {
+class StorageGridServiceReactive(
```

Comment:
Er det en idé å dele opp denne? F.eks. én service for hvert SG-api?

Reply:
Har tenkt på det, men det vil bli veldig mange Service-klasser å lenke inn fra Controlleren. Jeg tenker at de genererte klassene/APIene fra Swagger ikke nødvendigvis skal styre inndelingen i Service-klasser.  Vi får se om refactoring er nødvendig når/om det kommer behov for flere endepunkter definert av Controllere.

Reply:
Vi kan ta diskusjonen senere.

Reply:
Metoden UserPoliciesController.provisionUserPolicies() inneholder inneholder det meste av forretningslogikk i applikasjonen. Burde controlleren delegere dette til en fasade-service?

Reply:
Jeg er ikke enig i at provisionUserPolicies inneholder særlig med forretningslogikk. Den orkestrerer og delegerer, men det er i StorageGridService at forretningsdetaljene som genererer policies og koblinger mellom objektene ligger.  Det er flere tilnærminger til slik arkitektur, og vi kan godt ta en diskusjon om dette.

Reply:
Foreslår at vi lar denne være i MVP-versjon. Det kan hende Refactoring blir naturlig ved senere utvidelser. Ok?